### DISCUSSION TITLE
Transform: Labels to fields

### DISCUSSION THREAD

**Discussion by maveec** [grafana/grafana-json-datasource#260] [Ideas]
The *Labels to fields* transform seems not to work in the plugin. I cannot select any of the fields as "no options are found" although one of the fields is set as string from the query. Would it be possible to support this transform?
![image](https://user-images.githubusercontent.com/42912320/140305962-65bf2f9d-e1cc-466f-9dd1-d64091f01aed.png)
![image](https://user-images.githubusercontent.com/42912320/140306070-00007587-1320-4537-a8c1-394a65827662.png)

**Comment by marcusolsson** [comment]
This plugin doesn't use labels, so there's nothing to transform. Could you share more about what you're trying to do?

**Comment by maveec** [comment]
This is what I was trying to achieve. Basically to pivot the data with respect to a Field. We managed to overcome this issue simply by breaking the api calls per each different label and renaming the single query names. However it's a feature that could be handy. I am not sure whether it is feasible to allow a Field to be set as Label Type from within the plugin.
![image](https://user-images.githubusercontent.com/42912320/140780814-dc4474d3-ae5a-4cdb-a915-3435d336e96a.png)

**Reply by marcusolsson** [reply]
Sounds like you want to convert from Long to Wide format. If you're using a recent version of Grafana, the Prepare time series transform should be able to help here. Let me know if not!
